File: src/main/java/com/linebot/entity/Tobuy.java
package com.linebot.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.Date;

@Data
@Entity
@Table(name = "tobuy")
@NoArgsConstructor
@AllArgsConstructor
public class Tobuy {
	@Id
	@GeneratedValue(strategy= GenerationType.IDENTITY)
	@Column(name = "id")
	private Integer id;
	@Column(name = "goods")
	private String goods;
	@Column(name = "is_completed")
	private String isCompleted;
	@Column(name = "created_date")
	private Date createdDate;
	@Column(name = "updated_date")
	private Date updatedDate;

	// postgres は列追加の場所を指定できない
	@Column(name = "user_id")
	private String userId;
}

File: src/main/kotlin/com/linebot/entity/Tobuy.kt
package com.linebot.entity

import java.util.Date
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.Table

@Entity
@Table(name = "tobuy")
data class Tobuy(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    var id: Int? = null,
    @Column(name = "goods")
    var goods: String? = null,
    @Column(name = "is_completed")
    var isCompleted: String? = null,
    @Column(name = "created_date")
    var createdDate: Date? = null,
    @Column(name = "updated_date")
    var updatedDate: Date? = null,
    // postgres は列追加の場所を指定できない
    @Column(name = "user_id")
    var userId: String? = null
) {
    fun setIsCompleted(isCompleted: String?) = apply { this.isCompleted = isCompleted }
}
